src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1690,6 +1690,80 @@ void MacroAssembler::store_sized_value(Address dst, Register src, size_t size_in
   }
 }
 
+void MacroAssembler::load_int_misaligned(Register dst, Address src, Register tmp, bool is_signed) {
```

Comment:
`load_long_misaligned` provides `granularity`, maybe we add this to `load_int_misaligned` too? If granularity is 2, we can just use two `lh`s to load an int.